### PR TITLE
[None][fix] Reorder generation_logits to align with final beam search output ordering

### PR DESCRIPTION
**Pull request by achartier**
## Summary by CodeRabbit

* **Bug Fixes**
  * Generation logits are now correctly aligned with final beam search output ordering.
  * Updated guidance messaging for beam search with generation logits in streaming mode.

* **Tests**
  * Added validation test for generation logits accuracy with beam search decoding.

## Description

`generation_logits` are stored raw during generation - indexed by physical slot, not by beam identity. If the beam search is causing a reassignment, the end-user has no way of mapping the `generation_logits` at a given step to its beam.

This change re-orders the returned `generation_logits` based on which beam they are linked to at every generation step.

It builds a slot trace for each final beam, tracing backwards through `parentsId` to find which slot it was in at each step. This builds the slot list post-reassignment.

The slot list is then converted to the pre-reassignment slot using the `parentIds` data.

If logits reorder is needed, for each step g, logits are copied into a temporary buffer with the correct order, then copied back into the logits tensor.

## Test Coverage

New test test_whisper_beam_search_generation_logits

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: cpp/tensorrt_llm/batch_manager/trtGptModelInflightBatching.cpp [hunk 1]
```diff
@@ -1955,6 +1956,187 @@ void TrtGptModelInflightBatching::postProcessRequest(
...
+                }
+            }
+
+            for (SizeType32 g = 0; g < maxGenLen; ++g)
+            {
+                // Check if any beam needs reordering at this step
+                bool stepNeedsReorder = false;
+                for (SizeType32 b = 0; b < reqBeamWidth; ++b)
+                {
+                    if (slotTrace[b][g] != b)
```
Comment (pcastonguay) [inline cpp/tensorrt_llm/batch_manager/trtGptModelInflightBatching.cpp]: Here it seems like we would end up using `slotTrace[b][g]` for `g > genLen_b`, which will have 0-initialized value. Let's say for `b==1`, this could wrongly set `stepNeedsReorder` to true. Flagged by Claude but I think it makes sense.

**Review by pcastonguay — APPROVED** [review]

**Review by stnie — COMMENTED** [review]
I only reviewed `trtGptModelInflightBatching.cpp`  so far.  

Is this reordering of generation logits supported in the PyTorch workflow? If not are there plans to introduce it there as well?

**Review by stnie — COMMENTED** [review]
Two more comments on the tests

**Review by stnie — APPROVED** [review]

**Review by Funatiq — APPROVED** [review]
